tools/codegen/dest/native_functions.py

```diff
@@ -53,3 +53,49 @@ def compute_native_function_declaration(
     else:
         x = gen_unstructured(g, backend_index)
         return [] if x is None else [x]
+
+
+@with_native_function_and_index
+def gen_unstructured_lazy_definition(f: NativeFunction, backend_index: BackendIndex) -> Optional[str]:
+    sig = kernel_signature(f, backend_index)
+    metadata = backend_index.get_kernel(f)
+    
+    # Lazy IR stuff
+    all_types, value_types, scalar_types = process_ir_types(f.func)
+
+    lazy_tensors = "\n    ".join([f"LazyTensor l_{t.name} = bridge::GetLtcTensor({t.name});" for t in value_types])
+    node_ctor_inputs = ", ".join([f"l_{t.name}" for t in value_types] +
+                                 [s.name for s in scalar_types])
+    assert len(value_types) > 0, f"Only supporting tensor ops so far, none found in {sig}"
+    first_tensor = value_types[0]
+
+    if metadata is None:
+        return None
+    if "legacy::" in metadata.kernel:
+        return None
+    else:
+        return f"""\
+{sig.decl(name=metadata.kernel)} {{
+    {lazy_tensors}
+    return bridge::AtenFromLtcTensor(l_{first_tensor.name}.CreateFrom(
+        ir::MakeNode<ir::ops::{metadata.kernel}>({node_ctor_inputs})));
```

Comment:
It probably would make more sense to stop using `ir::MakeNode` and just arrange for the constructor on the IR class to be codegen friendly

Reply:
Absent that, this is a good use case for `tools.codegen.api.translate`. If you can define a signature that describes what the types of the arguments ir::MakeNode is looking for, you can use translate to automatically convert from the source arguments to the target arguments, handling reordering and conversions automatically

Reply:
so, ir::MakeNode is really just a wrapper around making a shared pointer.  I think what's more onerous is the pattern that we use the first tensor's lazytensor.CreateFrom() method.  I'm wondering if we can get rid of this and just create the node and wrap it in a new LazyTensor.